core/tests/webdriverio_utils/action.js

```diff
@@ -55,6 +55,14 @@ var getText = async function(elementName, element) {
   return await element.getText();
 };
 
+var waitForNonEmptyText = async function(elementName, element) {
```

Comment:
This entire function should be moved to waitFor,js.

Reply:
This function needs the `action.getText` function. If we move it to waitFor.js, we'll have a circular dependency: waitFor.js->action.js->waitFor.js

Reply:
You can just put something like the following in waitFor.js, right?

```
var waitForNonEmptyText = async function(element) {
  element.waitUntil(async function() {
     await waitFor.visibilityOf(element);
     return await element.getText();
  }, {
    timeout: DEFAULT_WAIT_TIME_MSECS,
  });
```

Reply:
Argh, dumb me, yeah, thanks